pypeit/core/extract.py

```diff
@@ -2386,3 +2387,174 @@ def ech_objfind(image, ivar, slitmask, slit_left, slit_righ, order_vec, maskslit
 
     return sobjs_final, skymask[allmask]
 
+
+class ManualExtractionObj(pypeitpar.ParSet):
+    """
+    An object holding the arguments for how to perform the
+    manual extraction of a spectrum.
+
+    A list of these objects is generated in pypeit.py
+    to perform a set of user-defined extractions.
+
+    For an example of how to define a series of manual extractions in
+    the pypeit input file, see :ref:`pypeit_file`.
+
+    Args:
+        frame (:obj:`str`):
+            The name of the fits file for a manual extraction
+        spat (list): List of spatial, spectral positions to hand extract
+            e.g. "1243.3:1200," or "1243.3:1200,1345:1200'
+        spec (list): List of spatial, spectral positions to hand extract
+            e.g. "1243.3:1200," or "1243.3:1200,1345:1200'
+        det (list): List of detectors for hand extraction. This must be a list aligned with spec and spat lists, or a single integer
+             which will be used for all members of that list
+        fwhm (list): List of FWHM for hand extraction. This must be a list aligned with spec and spat lists, or a single number which will
+             be used for all members of that list'
+
+
+    """
+    @classmethod
+    def by_fitstbl_input(cls, frame: str, inp: str):
+        """Generate the object from an entry in the fitstbl
+
+        Args:
+            frame (str): filename
+            inp (str): det:spat:spec:fwhm
+                1:1181.8:3820.6:3.
+
+        Returns:
+            ManualExtractionObj or None:
+        """
+        if len(inp.strip()) == 0:
+            return None
+
+        # Generate a dict
+        idict = dict(frame=frame, spat=[], spec=[], det=[], fwhm=[])
+        m_es = inp.split(',')
+        for m_e in m_es:
+            parse = m_e.split(':')
+            idict['det'] += [int(parse[0])]
+            idict['spat'] += [float(parse[1])]
+            idict['spec'] += [float(parse[2])]
+            idict['fwhm'] += [float(parse[3])]
+
+        # Build me
+        return cls.from_dict(idict)
+
+
+    def __init__(self, frame=None, spat=None, spec=None, det=None, fwhm=None):
+
+        # Grab the parameter names and values from the function
+        # arguments
+        args, _, _, values = pypeitpar.inspect.getargvalues(pypeitpar.inspect.currentframe())
```

Comment:
Hopefully we don't do this other places, but I think it's bad form to use packages imported by other modules as if they were attributes of that module.  I.e., don't use `pypeitpar.inspect`, reimport `inspect` at the top of this module and use it directly.  Same thing with `OrderedDict`.